From 87aa2520989bafbd861a03a32d6fdb0d2620faf5 Mon Sep 17 00:00:00 2001
From: Mart Raudsepp <leio@gentoo.org>
Date: Mon, 11 Mar 2019 23:23:11 +0200
Subject: [PATCH] Sync gnome-desktop code

Brings in fontconfig cache bind mount and a small memory fix; cache path hardcoded to Gentoo for now
---
 src/gnome-desktop/gnome-desktop-thumbnail-script.c | 4 ++++
 src/gnome-desktop/gnome-desktop-thumbnail.c        | 6 ++++--
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/gnome-desktop/gnome-desktop-thumbnail-script.c b/src/gnome-desktop/gnome-desktop-thumbnail-script.c
index 525766388..0331cb3c7 100644
--- a/src/gnome-desktop/gnome-desktop-thumbnail-script.c
+++ b/src/gnome-desktop/gnome-desktop-thumbnail-script.c
@@ -565,6 +565,10 @@ add_bwrap (GPtrArray   *array,
         }
     }
 
+  /* fontconfig cache if necessary */
+  if (!g_str_has_prefix ("/var/cache/fontconfig", "/usr/"))
+    add_args (array, "--ro-bind-try", "/var/cache/fontconfig", "/var/cache/fontconfig", NULL);
+
   add_args (array,
 	    "--proc", "/proc",
 	    "--dev", "/dev",
diff --git a/src/gnome-desktop/gnome-desktop-thumbnail.c b/src/gnome-desktop/gnome-desktop-thumbnail.c
index f48fd461a..566fbeb84 100644
--- a/src/gnome-desktop/gnome-desktop-thumbnail.c
+++ b/src/gnome-desktop/gnome-desktop-thumbnail.c
@@ -967,8 +967,10 @@ get_preview_thumbnail (const char *uri,
     if (file_info == NULL)
       return NULL;
 
-    object = g_object_ref (g_file_info_get_attribute_object (file_info,
-                                                             G_FILE_ATTRIBUTE_PREVIEW_ICON));
+    object = g_file_info_get_attribute_object (file_info,
+                                               G_FILE_ATTRIBUTE_PREVIEW_ICON);
+    if (object)
+        g_object_ref (object);
     g_object_unref (file_info);
 
     if (!object)
-- 
2.17.0

